shared/store/action-logger.js

```diff
@@ -15,13 +22,47 @@ const objToJS = state => {
   return newState
 }
 
-export const actionLogger = store => next => action => {
+const loggingStatSink: StatSink = {
+  label: 'Action Logger',
+  totalTime: 0,
+  totalActions: 0,
+  timings: [], // to calculate variance
+  startTime: 0,
+}
+
+const actionStatSink: StatSink = {
+  label: 'Actions',
+  totalTime: 0,
+  totalActions: 0,
+  timings: [], // to calculate variance
+  startTime: 0,
+}
+
+export const actionLogger = (store: any) => (next: any) => (action: any) => { // eslint-disable-line arrow-parens
   console.groupCollapsed && console.groupCollapsed(`Dispatching action: ${action.type}`)
 
   console.log(`Dispatching action: ${action.type}: ${JSON.stringify(action)} `)
-  let result = next(action)
 
-  console.log('Next state:', JSON.stringify(objToJS(store.getState())))
+  const shouldRunLogStats = shouldRunStats(logStatFrequency)
+  const shouldRunActionStats = shouldRunStats(actionStatFrequency)
+
+  const oldState = store.getState()
+
+  startTiming(shouldRunActionStats, actionStatSink)
+  const result = next(action)
+  endTiming(shouldRunActionStats, actionStatSink)
+
+  const newState = store.getState()
+
+  startTiming(shouldRunLogStats, loggingStatSink)
+  const diff = deep.diff(objToJS(oldState), objToJS(newState))
+  console.log('Diff:', JSON.stringify(diff))
```

Comment:
`console.log('Diff:', diff)` too...

I guess we stringify so it's a single line in log file.. but debugging... is there any way to have the file logging do stringify-ing?